src/screens/TopicPageScreen/index.js

```diff
@@ -59,7 +59,9 @@ const TopicPageScreen = (props) => {
   ).current;
 
   const [feedsContext, dispatch] = React.useContext(Context).feeds;
-  const feeds = feedsContext.topicFeeds;
+  const feeds = feedsContext.topicFeeds
+    ? feedsContext.topicFeeds.filter((feed) => feed?.topics?.includes(topicName))
+    : [];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The filtering operation on `feedsContext.topicFeeds` could potentially be expensive if the number of feeds is large. Consider using a more efficient data structure for `topics` such as a Set to improve lookup times from O(n) to O(1). However, this would require changes in how you store and retrieve topics.
```diff
- const feeds = feedsContext.topicFeeds
-   ? feedsContext.topicFeeds.filter((feed) => feed?.topics?.includes(topicName))
-   : [];
+ const feeds = feedsContext.topicFeeds
+   ? feedsContext.topicFeeds.filter((feed) => new Set(feed?.topics)?.has(topicName))
+   : [];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->